### PR TITLE
chore/many Resolve some let _ = statements

### PR DESCRIPTION
**Pull request by dirvine**
@fizyk20  This is a placeholder right now and needs debated perhaps @afck will need to confirm this is Ok to accept. We can discuss in the morning hopefully. Just trying to ensure we have not swallowed options and result types anywhere without at least a log statement.

### CODE REVIEW

File: src/ack_manager.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use error::RoutingError;
 use maidsafe_utilities::{self, serialisation};
 use message_filter::MessageFilter;
 use messages::RoutingMessage;
+use sha3;
```
Comment (afck) [inline src/ack_manager.rs:22]: I think features like this should ideally be in a separate commit, but please at least mention it in the commit message. :smile:

File: src/ack_manager.rs [hunk 2]
```diff
@@ -121,18 +129,24 @@ impl AckManager {
 impl Ack {
     pub fn compute(routing_msg: &RoutingMessage) -> Result<Ack, RoutingError> {
         let hash_msg = serialisation::serialise(routing_msg)?;
-        Ok(Ack(maidsafe_utilities::big_endian_sip_hash(&hash_msg)))
+        Ok(Ack {
+            id: maidsafe_utilities::big_endian_sip_hash(&hash_msg),
```
Comment (afck) [inline src/ack_manager.rs:133]: We could apply this one only to the SHA3 hash: Then we wouldn't have to go through the whole message twice.

File: src/ack_manager.rs [hunk 3]
```diff
@@ -121,18 +129,24 @@ impl AckManager {
...
+            m_hash: sha3::hash(&hash_msg),
+        })
     }
 }
 
 impl fmt::Display for Ack {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-        write!(formatter, "{:016x}", self.0)
+        write!(formatter, "id : {:016x}, hash : {:?}", self.id, self.m_hash)
```
Comment (afck) [inline src/ack_manager.rs]: The output would be less noisy if we'd print the hash as a hexadecimal number, too. (Also in `Debug`.)

File: src/data/priv_appendable_data.rs [hunk 4]
```diff
@@ -140,7 +140,11 @@ impl PrivAppendableData {
         self.owners = other.owners;
         self.data.extend(other.data);
         for ad in &self.deleted_data {
-            let _ = self.data.remove(ad);
+            let removed = self.data
+                .remove(&ad.clone());
```
Comment (afck) [inline src/data/priv_appendable_data.rs]: Why is this `clone` necessary? Shouldn't `ad` already be a reference and not be consumed?
(I also think rustfmt would undo that line break.)

File: src/data/priv_appendable_data.rs [hunk 5]
```diff
@@ -140,7 +140,11 @@ impl PrivAppendableData {
...
+            if !removed {
+                debug!("Could not remove deleted data : {:?}", ad);
```
Comment (afck) [inline src/data/priv_appendable_data.rs]: I think in this case there's really no reason to print anything:
`deleted` contains everything that has been deleted in previous updates, and it's expected that most of the entries already aren't present in `data` anymore.
We basically want to `if self.data.contains(ad) { self.data.remove(ad); }` here, but I think there's no way to do that without getting a result.
(Same for the other appendable type.)

File: src/messages.rs [hunk 6]
```diff
@@ -259,7 +259,10 @@ impl SignedMessage {
                 .cloned()
                 .collect_vec();
             for irrelevant_id in irrelevant_ids {
-                let _ = self.signatures.remove(&irrelevant_id);
+                let _removed_sig = self.signatures
+                    .remove(&irrelevant_id)
+                    .ok_or(debug!("Tried to remove a signature we did not have : {:?}",
+                                  irrelevant_id));
```
Comment (afck) [inline src/messages.rs:265]: In this case I'd even say: This is obviously unreachable code, and that be quickly verified by reading only the body of this single method, so let's not bother adding more code here.

File: src/messages.rs [hunk 7]
```diff
@@ -345,7 +348,10 @@ impl SignedMessage {
             })
             .collect_vec();
         for invalid_signature in &invalid_signatures {
-            let _ = self.signatures.remove(invalid_signature);
+            let _ignore_removed = self.signatures
+                .remove(invalid_signature)
+                .ok_or(debug!("Tried to remove an invalid signature we do not have : {:?}",
+                              invalid_signature));
```
Comment (afck) [inline src/messages.rs:354]: Same here: We just obtained `invalid_signatures` by filtering the keys of `self.signature`, so this code is unreachable and that can be verified within this function only.

File: src/messages.rs [hunk 8]
```diff
@@ -1002,7 +1008,10 @@ impl UserMessageCache {
                -> Option<UserMessage> {
         {
             let entry = self.0.entry((hash, part_count)).or_insert_with(BTreeMap::new);
-            let _ = entry.insert(part_index, payload);
+            let _count = entry.insert(part_index.clone(), payload.clone())
+                .ok_or(debug!("Could not insert part index : {:?} with payload {:?}",
+                              part_index,
+                              payload));
```
Comment (afck) [inline src/messages.rs]: `insert` returns `Some` if the value was already there. `None` is expected here. This should just be `if entry.insert(…).is_some()`, and the log should say that we got a duplicate message part.

File: src/mock_crust/support.rs [hunk 9]
```diff
@@ -67,10 +67,11 @@ impl Network {
         let endpoint = self.gen_endpoint(opt_endpoint);
 
         let handle = ServiceHandle::new(self.clone(), config, endpoint);
-        let _ = self.0
+        let _handle = self.0
             .borrow_mut()
             .services
-            .insert(endpoint, Rc::downgrade(&handle.0));
+            .insert(endpoint, Rc::downgrade(&handle.0))
+            .ok_or(debug!("Could not insert service handle "));
```
Comment (afck) [inline src/mock_crust/support.rs]: Same here: `None` is expected, I think, and `Some` would indicate a duplicate insertion.

File: src/mock_crust/support.rs [hunk 10]
```diff
@@ -278,7 +281,9 @@ impl ServiceImpl {
     }
 
     pub fn whitelist_peer(&mut self, peer_id: PeerId) {
-        let _ = self.whitelist.insert(peer_id);
+        if !self.whitelist.insert(peer_id.clone()) {
+            debug!("Could not insert whitelist for peer : {:?}", peer_id);
```
Comment (afck) [inline src/mock_crust/support.rs]: `Could not insert` sounds like it wasn't inserted, where in fact it was just inserted _twice_.

File: src/ack_manager.rs (hunk 3 above)
Comment (dirvine) [reply src/ack_manager.rs]: Ok made this print first 2 chars and ... to reduce noise.

File: src/data/priv_appendable_data.rs (hunk 4 above)
Comment (dirvine) [reply src/data/priv_appendable_data.rs]: Nice catch, removed clone()

File: src/data/priv_appendable_data.rs (hunk 5 above)
Comment (dirvine) [reply src/data/priv_appendable_data.rs]: Agreed, done.

File: src/messages.rs (hunk 6 above)
Comment (dirvine) [reply src/messages.rs:265]: Yes this I Think would be good to have a quick chat about.

File: src/messages.rs (hunk 8 above)
Comment (dirvine) [reply src/messages.rs]: Agreed, changed to suit.

File: src/mock_crust/support.rs (hunk 9 above)
Comment (dirvine) [reply src/mock_crust/support.rs]: Agree

File: src/mock_crust/support.rs (hunk 10 above)
Comment (dirvine) [reply src/mock_crust/support.rs]: Agree

**Review by dirvine — COMMENTED** [review]
Several good points fixed, poss require a hangout to discuss a little bit about the rest.
